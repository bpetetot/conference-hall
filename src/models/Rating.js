class Rating {
  constructor(data = {}) {
    this.uid = data.uid
    this.feeling = data.feeling
    this.rating = data.rating
    this.updateTimestamp = data.updateTimestamp
  }
}

export const ratingConverter = {
  toFirestore(rating) {
    return { ...rating }
  },
  fromFirestore(snapshot, options) {
    const data = snapshot.data(options)
    return new Rating(data)
  },
}

export default Rating
