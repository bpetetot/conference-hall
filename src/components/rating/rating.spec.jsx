/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import Rating from './index.js'

describe('components/rating', () => {
  it('should render', () => {
    const onRate = jest.fn()
    const wrapper = shallow(<Rating
      rating={0}
      onRating={onRate}
    />)
    expect(wrapper.state('rating')).toEqual(0)
    expect(wrapper).toMatchSnapshot()
    wrapper.find('i.fa-heart-o').simulate('click')
    expect(onRate).toHaveBeenCalled()
  })
})
