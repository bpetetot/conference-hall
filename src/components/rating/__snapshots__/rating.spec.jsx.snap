// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/rating should call onClick by default 1`] = `
<div
  className="rating"
  onMouseLeave={[Function]}
>
  <Tooltip
    inline={false}
    placement="left"
    tooltip="Without opinion"
  >
    <i
      className="fa fa-2x fa-ban rating-noopinion-disable"
      onClick={[Function]}
      onMouseEnter={[Function]}
      role="button"
    />
  </Tooltip>
  <Tooltip
    inline={false}
    placement="top"
    tooltip="No way! (0)"
  >
    <i
      className="fa fa-2x bounce"
      onClick={[Function]}
      onMouseEnter={[Function]}
      role="button"
    />
  </Tooltip>
  <i
    className="fa fa-2x fa-star-o"
    key="1"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="button"
  />
  <i
    className="fa fa-2x fa-star-o"
    key="2"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="button"
  />
  <i
    className="fa fa-2x fa-star-o"
    key="3"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="button"
  />
  <i
    className="fa fa-2x fa-star-o"
    key="4"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="button"
  />
  <i
    className="fa fa-2x fa-star-o"
    key="5"
    onClick={[Function]}
    onMouseEnter={[Function]}
    role="button"
  />
  <Tooltip
    inline={false}
    placement="right"
    tooltip="I love it! (5)"
  >
    <i
      className="fa fa-2x fa-heart-o"
      onClick={[Function]}
      onMouseEnter={[Function]}
      role="button"
    />
  </Tooltip>
</div>
`;
